spec.bs

```diff
@@ -13,3 +13,538 @@ Group: patcg
+  Topics can be send in the HTTP header for {{WindowOrWorkerGlobalScope/fetch()}} request. The response header for a topics related request can specify whether the caller should to be recorded.
+
+  <h3 id="send-browsing-topics-header-boolean-associated-with-request-header">send browsing topics header boolean associated with Request</h3>
+  A [=request=] has an associated <dfn for=request>send browsing topics header boolean</dfn>. Unless stated otherwise it is unset.
+
+  <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
+
+  <h3 id="browsing-topics-attribute-in-request-init-header">browsingTopics attribute in RequestInit</h3>
+  The <a href="https://fetch.spec.whatwg.org/#requestinit">RequestInit</a> dictionary contains a browsingTopics attribute:
+
+  <pre class=idl>
+  dictionary RequestInit {
+    // existing attributes ...
+
+    boolean browsingTopics;
+  };
+  </pre>
+
+  <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
+
+  <h3 id="modification-to-request-constructor-steps-header">Modification to request constructor steps</h3>
+  The following step will be added to the <a href="https://fetch.spec.whatwg.org/#dom-request">new Request(input, init) constructor steps</a>, before step "Set this's [=Request/request=] to |request|":
+
+  1. If <var ignore=''>init</var>["{{RequestInit/browsingTopics}}"] <a for=map>exists</a>, then set |request|'s [=request/send browsing topics header boolean=] to it.
+
+  <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
+
+  <h3 id="the-sec-browsing-topics-http-request-header-header">The \`<code>Sec-Browsing-Topics</code>\` HTTP request header</h3>
+
+  This specification defines a \`<dfn export http-header><code>Sec-Browsing-Topics</code></dfn>\` HTTP request header. It is used to send the topics.
+
+  <h3 id="modification-to-http-network-or-cache-fetch-algorithm-header">Modification to HTTP-network-or-cache fetch algorithm</h3>
+  The following step will be added to the <a href="https://fetch.spec.whatwg.org/#concept-http-network-or-cache-fetch">HTTP-network-or-cache fetch</a> algorithm, before step "Modify |httpRequest|'s [=request/header list=] per HTTP. ...":
+
+  1. <a>Append or modify a request \`<code>Sec-Browsing-Topics</code>\` header</a> for |httpRequest|.
+
+  <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
+
+  <h3 id="append-or-modify-a-request-sec-browsing-topics-header-header">Append or modify a request `Sec-Browsing-Topics` header</h3>
+
+  <div algorithm>
+    To <dfn>append or modify a request \`<code>Sec-Browsing-Topics</code>\` header</dfn>, given a [=request=] |request|, run these steps:
+      1. If |request|'s [=request/send browsing topics header boolean=] is not true, then return.
+      1. [=header list/Delete=] \`<code>Sec-Browsing-Topics</code>\` from |request|’s [=header list=].
```

Comment:
I'm not sure this should be necessary. No other header-setting spec does this.

Reply:
The Client Hints spec has similar mechanism https://wicg.github.io/client-hints-infrastructure/#abstract-opdef-remove-client-hints-from-redirect-if-needed. Basically the header is handled on a per-(redirected)-request level. Pervious state shouldn’t persist across redirects.

Reply:
So the intent is that if `https://example.com/a` redirects to `https://example.com/b`, then `fetch("https://example.com/a", { browsingTopics: true })` should send different `Sec-Browsing-Topics: ...` values to `/a` vs. `/b`, based on variations like the timestamp, random 5% chance, etc.?

If so, including that context as a `<p class="note">` would be helpful.